src-input/duk_bi_object.c

```diff
@@ -255,6 +257,11 @@ DUK_INTERNAL duk_ret_t duk_bi_object_constructor_seal_freeze_shared(duk_context
 	h = duk_require_hobject_promote_mask(ctx, 0, DUK_TYPE_MASK_LIGHTFUNC | DUK_TYPE_MASK_BUFFER);
 #endif
 
+	/* ES6 Sections 19.1.2.5, 19.1.2.17 */
+	if (!duk_is_object(ctx, 0)) {
+		return 1;
+	}
```

Comment:
How should plain buffers and lightfuncs behave here? They mimic objects, so they're generally accept anywhere where objects are.